### ISSUE TITLE
[Android] Use nativescript-local-notifications for background notifications

### ISSUE THREAD

**Issue by butaminas** [EddyVerbruggen/nativescript-plugin-firebase#1528]
Is there any way I could use `nativescript-local-notifications` for background notifications?

Right now I don't pass notification information in the notification builder but pass my information as data and therefore, with the help of `onMessageReceivedCallback` I can use `nativescript-local-notifications` to show the notifications.

Unfortunately, when the message is received in the background, this doesn't work. The only way so far I could receive notifications in the background is to use the initial way of showing the notification (by passing the title and body in the notification builder).

FYI: I need to use `nativescript-local-notifications` in the first place is because on some devices (e.g. Xiaomi phones) all the notifications from this plugin go to the miscellaneous channel and have low priority. With `nativescript-local-notifications` I can set a custom channel and force a priority if I need to.

**Comment by krthush** [comment]
I've been struggling with this exact same issue, the notifications occur in the same way as you have mentioned (I either have the option of using local notifications by not including the title/body, but then this doesn't work when the application is closed OR I include the title/body, but then I cannot override the FCM notification using local notifications).

My reason for requiring to use local notifications is bit different, with local notifications I can use buttons with my notifications which is very much required for our application.

Would be great if there are any updates on this? @butaminas let me know if you find a work around?

**Comment by petem24** [comment]
Try this as your payload

`{
   "data":{
     "title":"Title",
     "body":"Body Text",
     "showWhenInForeground": true
   },
     "notification": {
        "title" : "Title",
        "body"  : "Body Text",
        "showWhenInForeground": true
       },
     "to": "token here"
}`

**Comment by andymartinwork** [comment]
I have the same problem. I want to localise my messages into different languages, and thought it was easier to do this in the app than on the server, but it doesn't seem possible in the background.

I know that in iOS / Android natively you can add a handler in the AppDelegate / an activity in the AndroidManifest which is run when a notification is sent in the background. I'm not sure this has been implemented in nativescript-plugin-firebase?

**Comment by krthush** [comment]
So do you guys think thats its impossible to use this package and do FCM notifications with buttons? I noticed that its only possible for iOS, but I would really like to be able to make it for Android as well.

**Comment by yonkoGH** [comment]
1- Send your message as data only with title ,body , bade , etc.. ,Ï
2- Create your own custom Firebase service the extends the FirebaseMessagingService so you could receive data messages even when the app is killed.
3 - in your custom service code use the received  title & body   to send a local notification using Eddy's plugin "Local-Notification plugin" .
4 - inside your custom service code you can also set the number of badges using "Nativescript-plugin-badge"

**Comment by butaminas** [comment]
> 1- Send your message as data only with title ,body , badges , etc.. ,
> 2- Create your own custom Firebase service that extends the FirebaseMessagingService that will be called even when the app is killed.
> 3 - in your custom service code use the received title & body to send a local notification using Eddy's plugin "Local-Notification plugin" .
> 4 - inside your custom service code you can also set the number of badges using "Nativescript-plugin-badge"

Can you elaborate on step 2? How do you create a custom Firebase service that extends the FirebaseMessagingService and that will be called even when the app is killed?

**Comment by manojdcoder** [comment]
Yes, the service is meant to run always in background. You will have to subclass FirebaseMessagingService, something similar to the [default service](https://github.com/EddyVerbruggen/nativescript-plugin-firebase/blob/master/src/platforms/android/libraryproject/firebase/src/main/java/org/nativescript/plugins/firebase/MyFirebaseMessagingService.java) from plugin, except it's written in Java you may choose to do it with JS/TS itself. Refer the official marshalling docs for more info.

**Comment by andymartinwork** [comment]
It's important to note, this service will only work on Android.

**Comment by faisalansari20** [comment]
I am having similar issue. Is there any fix or work around available for both the platform?
